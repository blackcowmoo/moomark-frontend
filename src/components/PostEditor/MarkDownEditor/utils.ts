function canManipulateViaTextNodes(targetTextArea: HTMLTextAreaElement | HTMLInputElement) {
  if (targetTextArea.nodeName !== 'TEXTAREA') {
    return false;
  }
  let browserSupportsTextareaTextNodes;
  if (typeof browserSupportsTextareaTextNodes === 'undefined') {
    const textarea = document.createElement('textarea');
    textarea.value = '1';
    browserSupportsTextareaTextNodes = !!textarea.firstChild;
  }
  return browserSupportsTextareaTextNodes;
}

export function insertText(targetTextArea: HTMLTextAreaElement | HTMLInputElement, text: string) {
  targetTextArea.focus();

  // IE 8-10
  if ((document as any).selection) {
    const ieRange = (document as any).selection.createRange();
    ieRange.text = text;

    ieRange.collapse(false /* to the end */);
    ieRange.select();

    return;
  }

  // Webkit + Edge
  const isSuccess = document.execCommand('insertText', false, text);
  if (!isSuccess) {
    const start = targetTextArea.selectionStart;
    const end = targetTextArea.selectionEnd;
    if (!(start && end)) return;
    // Firefox (non-standard method)
    if (typeof (targetTextArea as any).setRangeText === 'function') {
      (targetTextArea as any).setRangeText(text);
    } else if (canManipulateViaTextNodes(targetTextArea)) {
      const textNode = document.createTextNode(text);
      let node = targetTextArea.firstChild;

      // If textarea is empty, just insert the text
      if (!node) {
        targetTextArea.appendChild(textNode);
      } else {
        // Otherwise we need to find a nodes for start and end
        let offset = 0;
        let startNode = null;
        let endNode = null;

        // To make a change we just need a Range, not a Selection
        const range = document.createRange();

        while (node && (startNode === null || endNode === null)) {
          const nodeLength = node.nodeValue ? node.nodeValue.length
            : 0;

          // if start of the selection falls into current node
          if (start >= offset && start <= offset + nodeLength) {
            range.setStart((startNode = node), start - offset);
          }

          // if end of the selection falls into current node
          if (end >= offset && end <= offset + nodeLength) {
            range.setEnd((endNode = node), end - offset);
          }

          offset += nodeLength;
          node = node.nextSibling;
        }

        if (start !== end) {
          range.deleteContents();
        }
        range.insertNode(textNode);
      }
    } else {
      const { value } = targetTextArea;
      targetTextArea.value = value.slice(0, start) + text + value.slice(end);
    }

    targetTextArea.setSelectionRange(start + text.length, start + text.length);

    const e = document.createEvent('UIEvent');
    e.initEvent('targetTextArea', true, false);
    targetTextArea.dispatchEvent(e);
  }
}
