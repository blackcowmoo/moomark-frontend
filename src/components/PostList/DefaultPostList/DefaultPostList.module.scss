@import '@styles/mixin';
@import '@styles/utils';

.container {
  @include mobile {
    width: calc(100% - 2rem);
  }
  width: 25rem;
  border-radius: 4px;
  background-color: var(--bg-primary);
  box-shadow: var(--box-shadow);
  margin: 1rem;
  .listTitle {
    border-bottom: solid 1px var(--text-secondary);
    padding: 0 1rem;
    margin: 0 1rem;
    // width: 80%;
  }
  ul {
    padding: 5px 0px 5px 5px;
    list-style: none;
    margin: 0px;
    padding: 0px;
    width: 100%;
  }
}

.DefaultViewItem {
  vertical-align: middle;
  margin: 0 1rem;
  border-bottom: solid 1px var(--text-secondary);
  .listItemWrapper {
    display: flex;
    margin-bottom: 0.5rem;
    align-items: center;
    .title {
      a {
        color: var(--text-primary);
      }
      width: 5rem;
      flex-grow: 4;
      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;
    }
    .postInfo {
      display: flex;
      flex-direction: row;
      .author {
        a {
          color: var(--text-primary);
        }
        margin-left: 1rem;
        flex-grow: 1;
        text-overflow: ellipsis;
        white-space: nowrap;
        overflow: hidden;
      }
      .like {
        margin-left: 1rem;
        flex-grow: 1;
        color: var(--accent);
      }
      .date {
        margin-left: 1rem;
        flex-grow: 3;
        color: $default-grey;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
      }
    }
  }
}

.TileViewItem {
  .listItemWrapper {
    .tileMain {
      .imgWrapper {
        img {
          width: 4rem;
          height: 4rem;
        }
      }
    }
  }
}
