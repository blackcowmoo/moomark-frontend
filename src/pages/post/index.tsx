import { CommentProps } from 'antd';
import PostViewer from '@components/PostViewer';

export interface PostProps{
  editorName: string;
  content: string;
  tags?: string[];
  comment?: CommentProps;
}

const PostPage: React.FC<PostProps> = () => {
  return (
    <>
      <PostViewer />
    </>
  );
};

export default PostPage;
