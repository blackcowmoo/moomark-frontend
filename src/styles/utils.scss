$global-background: #F5F5F5; 
$header-background: #ffffff;
$focus-brown: #ebc999;
// $default-green: #8ca9d3;
$default-green: #8DB48E;
$focus-red: #f26659;
$dark-green: #4D724D;
// $default-black: #000000ab;
$default-black: #373833;
$default-grey: rgb(173, 181, 189);