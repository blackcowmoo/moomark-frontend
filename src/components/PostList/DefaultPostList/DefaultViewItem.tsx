import Link from 'next/link';
import { IPostList } from 'types';
import { timeForToday } from 'utils/common';
import styles from './DefaultPostList.module.scss';

interface props {
  value: IPostList;
}

const DefaultViewItem: React.FC<props> = ({ value }) => {
  const { id, title, author, like, date } = value;
  return (
    <li className={styles.DefaultViewItem}>
      <h5 className={styles.listItemWrapper}>
        <div className={styles.title}>
          <Link href={`/post/${id}`}>{title}</Link>
        </div>
        <div className={styles.postInfo}>
          <div className={styles.like}>{like}</div>
          <div className={styles.author}>
            <Link href={`/user/${author}`}>{author}</Link>
          </div>
          <div className={styles.date}>{timeForToday(date)}</div>
        </div>
      </h5>
    </li>
  );
};

export default DefaultViewItem;
