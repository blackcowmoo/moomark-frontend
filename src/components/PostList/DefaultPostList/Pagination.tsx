import styles from './DefaultPostList.module.scss';

interface IPagination {
  totalPostCount: number;
  pageRange: number;
  currentPage: number;
  setPage: (input: number) => void;
}

const Pagination: React.FC<IPagination> = ({ totalPostCount, pageRange, currentPage, setPage }) => {
  const pageNumbers = [];
  for (let i = 1; i <= Math.ceil(totalPostCount / pageRange); i += 1) {
    pageNumbers.push(i);
  }

  const handlePaginate = (input: number) => {
    setPage(input);
  };

  return (
    <nav className={styles.Pagination}>
      {pageNumbers.map((number) => {
        return (
          <button className={currentPage === number ? styles.activePage : styles.page} key={number} onClick={() => handlePaginate(number)}>
            {number}
          </button>
        );
      })}
    </nav>
  );
};

export default Pagination;
