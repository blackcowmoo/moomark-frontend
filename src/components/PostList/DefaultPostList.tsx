import { PostListViewProps } from 'types';
import DefaultViewItem from './ListViewItem/DefaultViewItem';
import styles from './PostList.module.scss';

interface props {
  listTitle: string;
  postList: PostListViewProps[];
}

const DefaultPostList: React.FC<props> = ({ listTitle, postList }) => {
  return (
    <div className={styles.defaultPostList}>
      <h2 className={styles.listTitle}>{listTitle}</h2>
      <ul>
        {postList.map((value, index) => {
          return <DefaultViewItem value={value} key={index} />;
        })}
      </ul>
    </div>
  );
};

export default DefaultPostList;
