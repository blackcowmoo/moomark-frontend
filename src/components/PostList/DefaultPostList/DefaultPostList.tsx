import { PostListViewProps } from 'types';
import DefaultViewItem from './DefaultViewItem';
import styles from './DefaultPostList.module.scss';

export interface IDefaultPostList{
  listTitle: string;
  postList: PostListViewProps[];
}

const DefaultPostList: React.FC<IDefaultPostList> = ({ listTitle, postList }) => {
  return (
    <div className={styles.container}>
      <h2 className={styles.listTitle}>{listTitle}</h2>
      <ul>
        {postList.map((value, index) => {
          return <DefaultViewItem value={value} key={index} />;
        })}
      </ul>
    </div>
  );
};

export default DefaultPostList;
