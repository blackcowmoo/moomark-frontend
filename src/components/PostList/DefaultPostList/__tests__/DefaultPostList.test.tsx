import { mount } from 'enzyme';
import DefaultPostList from '@components/PostList/DefaultPostList';
import DefaultPostItem from '@components/PostList/DefaultPostList/DefaultPostItem';
import { HomePageListMock } from 'utils/mock';
import { IPostList } from 'types/post';

// const mockDate = new Date(2021, 1, 2);
const mockDefaultPostListItem: IPostList = { id: 1, title: '투자의 책임은 너에게', author: 'gosu', like: 222, uploadTime: '2021-1-2' };

describe('<DefaultPostList/>', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<DefaultPostList listTitle='테스트' postList={HomePageListMock} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});

describe('<DefaultPostItem />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<DefaultPostItem value={mockDefaultPostListItem} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('render userName', () => {
    const wrapper = mount(<DefaultPostItem value={mockDefaultPostListItem} />);
    expect(wrapper.props().value.id).toBe(1);
    expect(wrapper.props().value.title).toBe('투자의 책임은 너에게');
    expect(wrapper.props().value.author).toBe('gosu');
    expect(wrapper.props().value.like).toBe(222);
  });
});
