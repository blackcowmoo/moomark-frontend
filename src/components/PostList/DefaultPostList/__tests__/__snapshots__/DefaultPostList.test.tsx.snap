// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultPostItem /> matches snapshot 1`] = `
"<DefaultViewItem value={{...}}>
  <li className=\\"DefaultViewItem\\">
    <h5 className=\\"listItemWrapper\\">
      <div className=\\"title\\">
        <Link href=\\"/post/1\\">
          <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/1\\">
            투자의 책임은 너에게
          </a>
        </Link>
      </div>
      <div className=\\"postInfo\\">
        <div className=\\"like\\">
          222
        </div>
        <div className=\\"author\\">
          <Link href=\\"/user/gosu\\">
            <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
              gosu
            </a>
          </Link>
        </div>
        <div className=\\"date\\">
          2021년 1월 2일
        </div>
      </div>
    </h5>
  </li>
</DefaultViewItem>"
`;

exports[`<DefaultPostList/> matches snapshot 1`] = `
"<GridPostList listTitle=\\"테스트\\" postList={{...}}>
  <div className=\\"container\\">
    <h2 className=\\"listTitle\\">
      테스트
    </h2>
    <ul>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/1\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/1\\">
                  투자의 책임은 너에게
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/2\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/2\\">
                  버진은 날아오른다
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/blackCOw\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/blackCOw\\">
                    blackCOw
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/3\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/3\\">
                  숲을 먼저 보고 나무를 보아라.
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/4\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/4\\">
                  사는 것 보다 파는 것이 더 중요하다
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/의문의 주식고수\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/%EC%9D%98%EB%AC%B8%EC%9D%98%20%EC%A3%BC%EC%8B%9D%EA%B3%A0%EC%88%98\\">
                    의문의 주식고수
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/5\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/5\\">
                  ddd
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                1
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/6\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/6\\">
                  ddd
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/7\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/7\\">
                  ddd
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/8\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/8\\">
                  ddd
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/9\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/9\\">
                  ddd
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
      <DefaultViewItem value={{...}}>
        <li className=\\"DefaultViewItem\\">
          <h5 className=\\"listItemWrapper\\">
            <div className=\\"title\\">
              <Link href=\\"/post/10\\">
                <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/post/10\\">
                  ddd
                </a>
              </Link>
            </div>
            <div className=\\"postInfo\\">
              <div className=\\"like\\">
                222
              </div>
              <div className=\\"author\\">
                <Link href=\\"/user/gosu\\">
                  <a onClick={[Function: onClick]} onMouseEnter={[Function (anonymous)]} href=\\"/user/gosu\\">
                    gosu
                  </a>
                </Link>
              </div>
              <div className=\\"date\\">
                2021년 3월 10일
              </div>
            </div>
          </h5>
        </li>
      </DefaultViewItem>
    </ul>
  </div>
</GridPostList>"
`;
